### PR TITLE
fix: joinCall set correct mute state based on microphoneOn parameter

### PR DESCRIPTION
**Pull request by bfwg**
# What
Fixed a bug where the initial muted state is not being set correctly. 

# Why
When `microphoneOn` is true, `muted` will be true which is backward of what the code is intended to do. 

# How Tested
I tested calling `joinCall` with `microphoneOn` parameter `true`  and the call will start muted which is incorrect.

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts [hunk 1]
```diff
@@ -299,7 +299,7 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
     }
 
     return this.teeErrorToEventEmitter(() => {
-      const audioOptions: AudioOptions = { muted: microphoneOn ?? !this.getState().isLocalPreviewMicrophoneEnabled };
+      const audioOptions: AudioOptions = { muted: !microphoneOn ?? !this.getState().isLocalPreviewMicrophoneEnabled };
```
Comment (prprabhu-ms) [inline packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:302]: `microphoneOn` is optional. If we check for falsiness, calling `joinCall()` without any arguments will cause `muted` to be set to `true`.

So, fallback on `undefined`, and *then* check for falsiness...

```suggestion
      const audioOptions: AudioOptions = { muted: !(microphoneOn ?? this.getState().isLocalPreviewMicrophoneEnabled) };
```

**Review by prprabhu-ms — CHANGES_REQUESTED** [review]
Thank you for your contribution!

I checked in the rest of the codebase and turns out we don't ever call `joinCall` with `microphoneOn` set (to either `true` or `false`). That is why things were not on fire already.

Thanks for catching this.

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts (hunk 1 above)
Comment (bfwg) [reply packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts]: Great suggestion. I've fixed my code.

**Review by JamesBurnside — APPROVED** [review]

**Review by prprabhu-ms — APPROVED** [review]

**Comment by prprabhu-ms** [comment]
Do you mind running `rush changelog` and providing a changelog for this PR?

The type should be "patch" (as this is a bugfix). The PR title is a good change description.

**Comment by prprabhu-ms** [comment]
I'm trying to figure out why chromatic is unhappy.

**Comment by bfwg** [comment]
Done.

**Pull request merged**